packages/perspective/addon/components/perspective.ts

```diff
@@ -23,12 +26,31 @@ export default class PerspectiveVisualization extends YavinVisualizationComponen
     }
   }
 
+  async makeSchema(): Promise<Schema> {
+    const { response, request } = this.args;
+    // load some table data to infer a schema
+    const data = response.rows.slice(0, 10) as TableData;
+    const table = await worker.table(data);
+    const defaultSchema = await table.schema();
+    await table.delete();
+
+    const isDayAligned = (grain: Grain) => !['hour', 'minute', 'second'].includes(grain);
```

Comment:
I don't know if this system is capable of being timezone aware, but this isn't sufficient if you have contrasting arbitrary timezones.  There are 30 minute and 15 minute offset timezones.

Reply:
Based on side conversation, comment withdrawn.